src/duplicacy_snapshot.go

```diff
@@ -76,14 +76,24 @@ func CreateSnapshotFromDirectory(id string, top string) (snapshot *Snapshot, ski
                 continue
             }
 
-            if pattern[0] != '+' && pattern[0] != '-' {
+            if pattern[0] == '#' {
+                continue
+            }
+
+            if IsUnspecifiedFilter(pattern) {
                 pattern = "+" + pattern
             }
 
-            if pattern == "+" || pattern == "-" {
+            if IsEmptyFilter(pattern) {
                 continue
             }
 
+            valid, err := IsValidRegex(pattern)
+
+            if  !valid || err != nil {
+                LOG_ERROR("SNAPSHOT_FILTER", "Invalid regular express encountered for filter: \"%s\", error: %v", pattern, err)
```

Comment:
I've still got the same concern as before: I don't see how logging a problem with a single regex results in preventing all the rest from being (wrongly) applied. A set of unit tests (https://golang.org/pkg/testing/) would go a long way towards proving correct operation in this case.

Reply:
I believe you are now comfortable that LOG_ERROR terminates the program after logging the error. It's used throughout the duplicacy code, so I believe it's usage here is appropriate.

Reply:
That's correct, I have no remaining concerns about the changes in this pull request.